wp-content/plugins/goonj-blocks/build/render.php

```diff
@@ -40,10 +48,35 @@
 	$action_target['id']
 );
 
+$target_data = [
+	'dropping-center' => [
+		'start_time' => 'Dropping_Centre.Start_Time',
+		'end_time' => 'Dropping_Centre.End_Time',
+		'address' => 'Dropping_Centre.Where_do_you_wish_to_open_dropping_center_Address_',
+		'contribution_link' => $dropping_center_material_contribution_link,
+	],
+	'collection-camp' => [
+		'start_time' => 'Collection_Camp_Intent_Details.Start_Date',
+		'end_time' => 'Collection_Camp_Intent_Details.End_Date',
+		'address' => 'Collection_Camp_Intent_Details.Location_Area_of_camp',
+		'contribution_link' => $material_contribution_link,
+	],
+];
+
 if ( in_array( $target, array( 'collection-camp', 'dropping-center' ) ) ) :
-	$start_date = new DateTime( $action_target['Collection_Camp_Intent_Details.Start_Date'] );
-	$end_date   = new DateTime( $action_target['Collection_Camp_Intent_Details.End_Date'] );
-	$address = $action_target['Collection_Camp_Intent_Details.Location_Area_of_camp'];
+    $target_info = $target_data[$target];
+    
+    try {
+        $start_date = new DateTime($action_target[$target_info['start_time']]);
+        $end_date = new DateTime($action_target[$target_info['end_time']]);
+    } catch (Exception $e) {
+		Civi::log()->error("Error creating DateTime object: " . $e->getMessage());
+        $start_date = $end_date = new DateTime(); // Default to current date/time
```

Comment:
I expect the start and end dates are required fields, so why do we need the try-catch block here? Are there any cases where the start date or end date might be missing?

Reply:
Error handling is important here because, let say there is some case the required fields are mistakenly removed from the form builder, we need to implement error handling as a precautionary measure.